Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,244 @@
+
+    /**
+     * @inheritDoc
+     */
+    protected function createDriver(): QueueDriverInterface
+    {
+        return $this->getContainer()->get('smartesb.drivers.queue.amqp');
+    }
+
+    /**
+     * Returns an instance of the AsyncQueueConsumer class
+     *
+     * @return AsyncQueueConsumer
+     */
+    protected function createConsumer()
+    {
+        return $this->getContainer()->get('smartesb.async_consumers.queue');
+    }
+
+    /**
+     * Test the connection set in the driver initialization
+     *
+     * @group amqp-connection
+     */
+    public function testConnection()
+    {
+        $this->assertInstanceOf(AMQPStreamConnection::class, $this->driver->declareChannel()->getConnection());
+        $this->assertTrue($this->driver->isConnected());
+    }
+
+    /**
+     * Tests the normal process of publish messages
+     *
+     * @dataProvider getMessages
+     * @param MessageInterface $msg
+     * @throws \Exception
+     * @group amqp-send
+     */
+    public function testSend(MessageInterface $msg)
+    {
+        $msgIn = $this->createQueueMessage($msg);
+        $msgIn->addHeader('test_header', '12345');
+        $this->assertTrue($this->driver->send($msgIn));
+    }
+
+    /**
+     * Test the consume process without a callback in the return
+     *
+     * @group amqp-consume-no-callback
+     */
+    public function testConsumeWithoutCallback()
+    {
+        $this->consumer = $this->createConsumer();
+        $consumerTag = $this->consumer->getName();
+        $this->driver->declareChannel();
+        $this->driver->declareQueue($this->queueName, AMQP_DURABLE, []);
+        $return = $this->driver->consume($consumerTag, $this->queueName);
+        $this->assertNull($return);
+    }
+
+    /**
+     * Tests the consume process with a callback in return that receives the messages and ack this after
+     *
+     * @dataProvider getMessages
+     * @group amqp-consume-callback
+     */
+    public function testConsumeWithCallbackAckingMessage(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+
+        $amqpMessage = null;
+        $callback = function(AMQPMessage $message) use (&$amqpMessage){
+            $amqpMessage = $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($amqpMessage->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->isConsuming();
+        $this->driver->wait();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertInstanceOf(AsyncQueueConsumer::class, $this->consumer);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertContains('QueueMessage', $amqpMessage->getBody());
```

Comment:
Maybe a stupid question, how can we consume and receive a message if we didn't send any message? Or you are relying on the message inserted of the `testSend` message?